enhancements/local-storage/automatic-partition.md

```diff
@@ -0,0 +1,244 @@
+  
+  // TotalProvisionedPartitionCount is the count of the total partitions over which the PVs has been provisioned
+  // This will be diffrent for TotalProvisionedDeviceCount as TotalProvisionedDeviceCount is
+  //the count of the actual devices
+  TotalProvisionedPartitionCount *int32 `json:"totalProvisionedPartitionCount,omitempty"`
+  
+  // PartitioningStatus will track the status of partitioning
+  // Valid values are
+  // nil : if there is no partitioningSpec
+  // partitoining : if the partitioning is in progress
+  // completed: when the partitioning of devices is completed
+  PartitioningStatus string `json:"partitioningStatus"`
+}
+```
+
+### Workflow of `LocalPV` creation via `LocalVolumeSet CR`
+- Once the `LocalVolumeSet` is created, the controller will:
+  - configure the local-static-provisioner to make a new StorageClass based on certain directories on the selected nodes.
+  - assign diskmaker daemonset to the selected nodes.
+  - there will be one daemonset for each `LocalVolumeSet` CR.
+- The diskmaker daemon will find devices that match the disovery policy and check if partitoningSpec is present
+  - if partitonSpec is present it will set PartitioningStatus to partitioning and will partition the devices.
+  - as soon as a device is partitoned
+    - the diskmaker will symlink the child partitions of the device into the directory that the local-static-provisioner is watching
+    - will set partitioningStatus to Completed once all the devices are partitioned
+- The localVolumeset won't accept changes after its created and has a partitioningSpec
+
+Example of `LocalVolumeSet` CR:
+
+```yaml
+apiVersion: local.storage.openshift.io/v1alpha1
+kind: LocalVolumeSet
+metadata:
+  name: example-autodetect
+spec:
+  nodeSelector:
+    nodeSelectorTerms:
+      - matchExpressions:
+          - key: kubernetes.io/hostname
+            operator: In
+            values:
+              - worker-0
+              - worker-1
+  storageClassName: example-storageclass
+  volumeMode: Block
+  minDeviceCount: 5
+  maxDeviceCount: 10
+  deviceInclusionSpec:
+    deviceTypes:
+      - RawDisk
+    deviceMechanicalProperty:
+      - Rotational
+      - NonRotational
+    minSize: 10G
+    maxSize: 100G
+  partitioningSpec:
+    size: 30G
+status:
+  conditions:
+  - lastTransitionTime: "2020-03-17T09:33:43Z"
+    status: "True"
+    type: Available
+  totalProvisionedDeviceCount: 5
+  totalProvisionedPartitionCount: 15
+  partitioningStatus: completed
+```
+
+As per the above example the administrator wants partitions of size 30G.
+Lets assume that there are 5 devices of 100G  that are detected and are fit for partitoning, so each device will be partitioned into 3 parts of 30G each and 10G of each device will remain unpartitioned.
+Right now we are keeping it simple by partitioning by size and will move to advanced partitioning policy in the further releases to avoid having unusable disk space.
```

Comment:
I would think the admins would most often carve up the devices in sizes that make sense for their disks, so this may not even be a common issue. Sounds good for a future release though if it's common enough.